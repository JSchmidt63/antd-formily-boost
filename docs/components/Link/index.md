---
title: Link组件
order: 3
nav:
    title: 组件
group:
    title: Link组件
    order: 1
---

## 基础使用

<code src="./link.tsx"></code>

这个组件就没啥好说的吧

<code src="./label.tsx"></code>

<API src="../../../src/Link.tsx">SpaceDivider</API>
