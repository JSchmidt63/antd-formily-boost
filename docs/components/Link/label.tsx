import { createForm, onFieldReact } from '@formily/core';
import { createSchemaField, FormConsumer, Schema } from '@formily/react';
import { Label } from 'antd-formily-boost';
import { Form } from '@formily/antd';
import { useMemo } from 'react';
import { observable } from '@formily/reactive';
import React from 'react';

const SchemaField = createSchemaField({
    components: {
        Label,
    },
});

let lastState = observable({
    data: [],
});

export default () => {
    const form = useMemo(() => {
        return createForm({
            values: lastState,
        });
    }, []);
    return (
        <Form form={form} feedbackLayout="terse">
            <SchemaField>
                <SchemaField.String
                    name="to"
                    title="连接"
                    x-component={'Label'}
                />
            </SchemaField>

            <FormConsumer>
                {() => <div>{JSON.stringify(form.values)}</div>}
            </FormConsumer>
        </Form>
    );
};
