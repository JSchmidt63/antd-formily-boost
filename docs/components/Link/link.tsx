import { createForm, onFieldReact } from '@formily/core';
import { createSchemaField, FormConsumer, Schema } from '@formily/react';
import { Link } from 'formily-antd';
import { Form } from '@formily/antd';
import { useMemo } from 'react';
import { observable } from '@formily/reactive';

const SchemaField = createSchemaField({
    components: {
        Link,
    },
});

let lastState = observable({
    data: [],
});

export default () => {
    const form = useMemo(() => {
        return createForm({
            values: lastState,
        });
    }, []);
    return (
        <Form form={form} feedbackLayout="terse">
            <SchemaField>
                <SchemaField.Void
                    name="to"
                    title="连接"
                    x-component={'Link'}
                    x-component-props={{
                        to: '123',
                    }}
                />
            </SchemaField>

            <FormConsumer>
                {() => <div>{JSON.stringify(form.values)}</div>}
            </FormConsumer>
        </Form>
    );
};
