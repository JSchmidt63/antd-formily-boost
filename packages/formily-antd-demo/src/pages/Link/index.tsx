import { createForm, onFieldInputValueChange } from '@formily/core';
import {
  createSchemaField,
  Field,
  FormConsumer,
  ObjectField,
} from '@formily/react';
import FormilyAntdTable from 'formily-antd';
import {
  Form,
  FormItem,
  Input,
  Select,
  PreviewText,
  Space,
} from '@formily/antd';
import ProCard from '@ant-design/pro-card';
import 'antd/dist/antd.css';

const form = createForm({
  initialValues: {
    data: [
      {
        name: 'fish',
        age: 123,
      },
      {
        name: 'cat',
        age: 456,
      },
      {
        name: 'dog',
        age: 789,
      },
    ],
  },
  effects: () => {},
});

const SchemaField = createSchemaField({
  components: {
    FormItem,
    Input,
    Select,
    Table: FormilyAntdTable,
  },
});

export default () => {
  return (
    <Space
      style={{
        background: 'rgb(240, 242, 245)',
        padding: '20px',
        display: 'flex',
      }}
      direction="vertical"
      size={10}
    >
      <ProCard title="基础">
        <Form form={form} feedbackLayout="terse">
          <SchemaField>
            <SchemaField.Array name="data" x-component="Table">
              <SchemaField.Void>
                <SchemaField.Void
                  name="firstColumn"
                  x-component="Table.Column"
                  x-component-props={{
                    title: '名字',
                    width: '50%',
                  }}
                >
                  <SchemaField.String name="name" x-component={'Input'} />
                </SchemaField.Void>

                <SchemaField.Void
                  x-component="Table.Column"
                  x-component-props={{
                    title: '年龄',
                    width: '50%',
                  }}
                >
                  <SchemaField.String name="age" x-component={'Table.Text'} />
                </SchemaField.Void>
              </SchemaField.Void>
            </SchemaField.Array>
          </SchemaField>
        </Form>
      </ProCard>

      <ProCard title="数据">
        <Form form={form} feedbackLayout="terse">
          <FormConsumer>
            {() => <div>{JSON.stringify(form.values)}</div>}
          </FormConsumer>
        </Form>
      </ProCard>
    </Space>
  );
};
