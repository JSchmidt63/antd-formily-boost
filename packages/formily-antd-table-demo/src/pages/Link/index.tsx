import { createForm, onFieldInputValueChange } from '@formily/core';
import {
  createSchemaField,
  Field,
  FormConsumer,
  ObjectField,
} from '@formily/react';
import FormilyAntdTable from 'formily-antd-table';
import { Form, FormItem, Input, Select, PreviewText } from '@formily/antd';
import ProCard from '@ant-design/pro-card';

const form = createForm({
  initialValues: {
    data: [
      {
        name: 'fish',
        age: 123,
      },
      {
        name: 'cat',
        age: 456,
      },
      {
        name: 'dog',
        age: 789,
      },
    ],
  },
  effects: () => {},
});

const SchemaField = createSchemaField({
  components: {
    FormItem,
    Input,
    Select,
    Table: FormilyAntdTable,
  },
});

export default () => {
  return (
    <div style={{ background: 'rgb(240, 242, 245)', padding: '20px' }}>
      <ProCard title="基础">
        <Form form={form} feedbackLayout="terse">
          <SchemaField>
            <SchemaField.Array name="data" x-component="Table">
              <SchemaField.Void>
                <SchemaField.Void
                  name="firstColumn"
                  x-component="Table.Column"
                  x-component-props={{
                    title: '名字',
                    width: '100px',
                  }}
                >
                  <SchemaField.String name="name" x-component={'Table.Text'} />
                </SchemaField.Void>

                <SchemaField.Void
                  x-component="Table.Column"
                  x-component-props={{
                    title: '年龄',
                  }}
                >
                  <SchemaField.String name="age" x-component={'Table.Text'} />
                </SchemaField.Void>
              </SchemaField.Void>
            </SchemaField.Array>
          </SchemaField>
          <FormConsumer>
            {() => <div>{JSON.stringify(form.values)}</div>}
          </FormConsumer>
        </Form>
      </ProCard>
    </div>
  );
};
