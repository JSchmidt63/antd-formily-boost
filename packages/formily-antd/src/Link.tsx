import React from 'React';
import { observer, useField } from '@formily/react';
import { useHistory } from 'umi';

type LinkToProps =
    | {
          pathname?: string;
          query?: object;
          state?: object;
      }
    | string;

type LinkProps = {
    to: LinkToProps;
    onClick?: () => void;
};

const MyLink: React.FC<LinkProps> = observer((props) => {
    const history = useHistory();
    const field = useField();
    let onClick = props.onClick
        ? props.onClick
        : () => {
              history.push(props.to);
          };
    return <a onClick={onClick}>{field.title}</a>;
});

export default MyLink;
