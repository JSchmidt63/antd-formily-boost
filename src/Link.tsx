import React from 'react';
import { observer, useField } from '@formily/react';

type LinkProps = {
    to: string;
    onClick?: () => void;
};

const MyLink: React.FC<LinkProps> = observer((props) => {
    const field = useField();
    let onClick = props.onClick
        ? props.onClick
        : () => {
              window.location.href = props.to;
          };
    return <a onClick={onClick}>{field.title}</a>;
});

export default MyLink;
