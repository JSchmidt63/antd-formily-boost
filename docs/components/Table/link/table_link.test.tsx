import Link from './link';
import LinkReact from './linkReact';
import { render } from '@testing-library/react';
import React from 'react';

beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
        value: jest.fn(() => {
            return {
                matches: true,
                addListener: jest.fn(),
                removeListener: jest.fn(),
            };
        }),
    });
});

test('Link', () => {
    const { asFragment } = render(<Link />);
    expect(asFragment()).toMatchSnapshot();
});

test('LinkReact', () => {
    const { asFragment } = render(<LinkReact />);
    expect(asFragment()).toMatchSnapshot();
});
