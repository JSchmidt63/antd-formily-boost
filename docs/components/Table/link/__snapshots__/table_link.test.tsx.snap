// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link 1`] = `
<DocumentFragment>
  <div
    class="ant-formily-layout"
  >
    <form>
      <div
        class="ant-table-wrapper antd_formily_table_boost undefined"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-bordered"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          名字
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          年龄
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          操作
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                            data-span="1"
                          >
                            <div
                              class="ant-formily-item-control"
                            >
                              <div
                                class="ant-formily-item-control-content"
                              >
                                <div
                                  class="ant-formily-item-control-content-component"
                                >
                                  <span
                                    class="ant-input-affix-wrapper"
                                  >
                                    <input
                                      class="ant-input"
                                      type="text"
                                      value="fish"
                                    />
                                    <span
                                      class="ant-input-suffix"
                                    >
                                      <span />
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          123
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <a>
                              编辑
                            </a>
                            <div
                              class="ant-divider ant-divider-vertical"
                              role="separator"
                            />
                            <a>
                              删除
                            </a>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                            data-span="1"
                          >
                            <div
                              class="ant-formily-item-control"
                            >
                              <div
                                class="ant-formily-item-control-content"
                              >
                                <div
                                  class="ant-formily-item-control-content-component"
                                >
                                  <span
                                    class="ant-input-affix-wrapper"
                                  >
                                    <input
                                      class="ant-input"
                                      type="text"
                                      value="cat"
                                    />
                                    <span
                                      class="ant-input-suffix"
                                    >
                                      <span />
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          456
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <a>
                              编辑
                            </a>
                            <div
                              class="ant-divider ant-divider-vertical"
                              role="separator"
                            />
                            <a>
                              删除
                            </a>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                            data-span="1"
                          >
                            <div
                              class="ant-formily-item-control"
                            >
                              <div
                                class="ant-formily-item-control-content"
                              >
                                <div
                                  class="ant-formily-item-control-content-component"
                                >
                                  <span
                                    class="ant-input-affix-wrapper"
                                  >
                                    <input
                                      class="ant-input"
                                      type="text"
                                      value="dog"
                                    />
                                    <span
                                      class="ant-input-suffix"
                                    >
                                      <span />
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          789
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <a>
                              编辑
                            </a>
                            <div
                              class="ant-divider ant-divider-vertical"
                              role="separator"
                            />
                            <a>
                              删除
                            </a>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        {"data":[{"name":"fish","age":123},{"name":"cat","age":456},{"name":"dog","age":789}]}
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`LinkReact 1`] = `
<DocumentFragment>
  <div
    class="ant-formily-layout"
  >
    <form>
      <div
        class="ant-table-wrapper antd_formily_table_boost undefined"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-bordered"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          名字
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          年龄
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          操作
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                            data-span="1"
                          >
                            <div
                              class="ant-formily-item-control"
                            >
                              <div
                                class="ant-formily-item-control-content"
                              >
                                <div
                                  class="ant-formily-item-control-content-component"
                                >
                                  <span
                                    class="ant-input-affix-wrapper"
                                  >
                                    <input
                                      class="ant-input"
                                      type="text"
                                      value="fish"
                                    />
                                    <span
                                      class="ant-input-suffix"
                                    >
                                      <span />
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          123
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <a>
                              编辑
                            </a>
                            <div
                              class="ant-divider ant-divider-vertical"
                              role="separator"
                            />
                            <a>
                              删除
                            </a>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                            data-span="1"
                          >
                            <div
                              class="ant-formily-item-control"
                            >
                              <div
                                class="ant-formily-item-control-content"
                              >
                                <div
                                  class="ant-formily-item-control-content-component"
                                >
                                  <span
                                    class="ant-input-affix-wrapper"
                                  >
                                    <input
                                      class="ant-input"
                                      type="text"
                                      value="cat"
                                    />
                                    <span
                                      class="ant-input-suffix"
                                    >
                                      <span />
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          456
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <a>
                              编辑
                            </a>
                            <div
                              class="ant-divider ant-divider-vertical"
                              role="separator"
                            />
                            <a>
                              删除
                            </a>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-terse ant-formily-item-label-align-right ant-formily-item-control-align-left"
                            data-span="1"
                          >
                            <div
                              class="ant-formily-item-control"
                            >
                              <div
                                class="ant-formily-item-control-content"
                              >
                                <div
                                  class="ant-formily-item-control-content-component"
                                >
                                  <span
                                    class="ant-input-affix-wrapper"
                                  >
                                    <input
                                      class="ant-input"
                                      type="text"
                                      value="dog"
                                    />
                                    <span
                                      class="ant-input-suffix"
                                    >
                                      <span />
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          789
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <a>
                              编辑
                            </a>
                            <div
                              class="ant-divider ant-divider-vertical"
                              role="separator"
                            />
                            <a>
                              删除
                            </a>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        {"data":[{"name":"fish","age":123},{"name":"cat","age":456},{"name":"dog","age":789}]}
      </div>
    </form>
  </div>
</DocumentFragment>
`;
